ppci/lang/c/semantics.py

```diff
@@ -320,17 +320,21 @@ def check_redeclaration_storage_class(self, sym, declaration):
         old_storage_class = sym.declaration.storage_class
         new_storage_class = declaration.storage_class
         # None == automatic storage class.
-        invalid_combos = [(None, "static"), ("extern", "static")]
+        # changes of linkage between internal and external are illegal
```

Comment:
Playing around a bit [at this site](https://godbolt.org/) makes it appear differently, you can declare a function as non-static after first specifying it as static.